src/common/popup/components/copy-button.test.jsx

```diff
@@ -1,44 +1,79 @@
 import { shallow } from 'enzyme';
-import React from 'react';
+import React, { useContext } from 'react';
 
 import EnvContext from '../env-context';
 import CopyButton from './copy-button';
 
+jest.mock('react', () => ({
+  ...jest.requireActual('react'),
+  useContext: jest.fn(),
+}));
+jest.useFakeTimers();
+
 describe('copy-button', () => {
   function render(overrides, env) {
+    useContext.mockReturnValue(env);
     const defaults = { value: 'copy text' };
-
     const props = { ...defaults, ...overrides };
-
-    // Enzyme does not fully support the new React Context API at the moment.
-    // We work around this limitation by directly passing in the context value
-    // to the `EnvContext.Consumer` child function for rendering.
-    //
-    // See https://github.com/airbnb/enzyme/issues/1553.
-    const outer = shallow(<CopyButton {...props} />);
-    const children = outer.find(EnvContext.Consumer).prop('children');
-    const wrapper = shallow(children(env));
-
-    return wrapper;
+    return shallow(<CopyButton {...props} />);
   }
 
-  let grab;
+  let pbcopy;
+  let close;
 
   beforeEach(() => {
-    grab = jest.fn();
+    useContext.mockReset();
+    pbcopy = jest.fn();
+    close = jest.fn();
+  });
+
+  afterEach(() => {
+    jest.clearAllTimers();
+  });
+
+  it('uses the env context', () => {
+    render({}, { pbcopy }); // render to check use of context
+    expect(useContext).toHaveBeenCalledWith(EnvContext);
+  });
+
+  it('renders its children', () => {
+    const children = 'button content';
+    const wrapper = render({ children }, { pbcopy });
+    expect(wrapper.contains(children)).toBe(true);
   });
 
-  it('calls the context grab function with the provided value on click', () => {
+  it('passes the "copied" state to its children if they are a function', () => {
+    const children = jest.fn();
+    const wrapper = render({ children }, { pbcopy });
+    expect(children).toHaveBeenNthCalledWith(1, false);
+
+    wrapper.simulate('click');
+
+    expect(children).toHaveBeenNthCalledWith(2, true);
+  });
+
+  it('calls the context pbcopy function with the provided value on click', () => {
     const value = 'a lot of value for such a small button';
-    const wrapper = render({ value }, { grab });
+    const wrapper = render({ value }, { pbcopy });
     wrapper.simulate('click');
-    expect(grab).toHaveBeenCalledWith(value);
+    expect(pbcopy).toHaveBeenCalledWith(value);
+  });
+
+  it('calls the context close function delayed after the value was copied', () => {
+    const value = 'a lot of value for such a small button';
+    const wrapper = render({ value }, { pbcopy, close });
+    wrapper.simulate('click');
+    expect(pbcopy).toHaveBeenCalledWith(value);
+    expect(close).not.toHaveBeenCalled();
+
+    jest.advanceTimersByTime(700);
```

Comment:
An alternative could be to use `jest.runOnlyPendingTimers()`, so the test is independent of the timeout delay. But I also like the way it's tested now. 🙂 

— https://jestjs.io/docs/en/timer-mocks#run-pending-timers

Reply:
I actually used this before. But making the test independent of the actual delay in ms is a good point. Will switch it back again.